src/Component.php

```diff
@@ -50,17 +43,14 @@ private function write(Client $client, Config $config, string $bucket) : void
             $primaryKey = $manifest['primary_key'] ?? [];
             $csv = new CsvFile($this->getDataDir() . '/in/tables/' . $table['destination']);
             $tableId = $bucket . '.' . $table['destination'];
-            if ($client->tableExists($tableId)) {
-                $tableInfo = $client->getTable($tableId);
-                if ($tableInfo['primaryKey'] != $primaryKey) {
-                    throw new UserException(
-                        'Primary in the destination table ' . $table['destination'] . ' ' .
-                        json_encode($tableInfo['primaryKey']) .
-                        ' does not match the primary key in the source table: ' . json_encode($primaryKey)
-                    );
+            if ($config->isFullSync()) {
+                try {
+                    $client->dropTable($tableId);
+                } catch (ClientException $e) {
+                    if ($e->getCode() !== 404) {
```

Comment:
není lepší místo toho použít `$client->tableExists($tableId)`? 404 může vyhodit třeba i neexistující bucket (jo, vím, odchytil by se v autorizaci). kdyby tam byla nějaký konkrétní výjimka, `TableNotFoundException`, tak by to šlo, ale takhle mi to přijde náchylní na false positive

Reply:
aha, ja nevim no - kdysi me nekdo rekl, ze to mam delat takhle, protoze jinak tableExists a dropTable neni atomicky. No - ono to neni atomicky ani o radek dal pri create, tak nevim

Reply:
souhlas, měl by tam bejt rovnou drop table a ten by měl vyhazovat správný exceptions. takhle je to blbý a s tím `$client->tableExists($tableId)` je to neatomický.